cmake/JuliaDeps.cmake

```diff
@@ -0,0 +1,16 @@
+jl_str_option(LLVM_VER "" 3.3)
+jl_str_option(LLVM_LIB_SUFFIX "" "")
+jl_str_option(PCRE_VER "" 10.10)
+jl_str_option(DSFMT_VER "" 2.2.3)
+jl_str_option(LAPACK_VER "" 3.5.0)
+jl_str_option(ARPACK_VER "" 3.2.0)
+jl_str_option(FFTW_VER "" 3.3.4)
+jl_str_option(SUITESPARSE_VER "" 4.4.2)
+jl_str_option(UNWIND_VER "" 1.1)
+jl_str_option(OSXUNWIND_VER "" 0.0.1)
+jl_str_option(GMP_VER "" 6.0.0)
+jl_str_option(MPFR_VER "" 3.1.2)
+jl_str_option(PATCHELF_VER "" 0.8)
+jl_str_option(GIT_VER "" 1.8.5.6)
+jl_str_option(VIRTUALENV_VER "" 1.11.6)
+jl_str_option(RMATH_JULIA_VER "" 0.1)
```

Comment:
Which logic would be duplicated? We don't exactly have anything that perfectly parallels `configure_file` in the current system. A few auto-populated headers and `jl` files but that's not quite the same thing.

It's not realistic to make the transition immediately mandatory, there will be a few missing pieces in the cmake setup at first that we'll have to have some hands on time using it to find. We should disable in-tree builds for cmake, and not break the makefiles until there's buy-in from enough people trying and using it for a while that it's a strict improvement and the makefiles can be completely deleted. Otherwise this will languish permanently as an open PR.